src/upchunk.spec.ts

```diff
@@ -1,333 +1,298 @@
-    );
-  });
-});
+    upload.on('chunkSuccess', () => {
+      calls = calls + 1;
+    });
 
-test('a single chunk failing the max number of times fails the upload', (done) => {
-  nock('https://example.com')
-    .put('/upload/endpoint')
-    .times(5)
-    .reply(502)
-    .put('/upload/endpoint')
-    .twice()
-    .reply(200);
-
-  const upload = createUploadFixture({ delayBeforeAttempt: 0.1 });
-
-  upload.on('error', (err) => {
-    try {
-      expect(err.detail.chunk).toBe(0);
-      expect(err.detail.attempts).toBe(5);
+    upload.on('success', () => {
+      expect(calls).to.equal(2);
       done();
-    } catch (err) {
-      done(err);
-    }
-  });
-
-  upload.on('success', () => {
-    done(`Expected upload to fail due to failed attempts`);
+    });
   });
-});
-
-test('chunkSuccess event is fired after each successful upload', (done) => {
-  nock('https://example.com')
-    .put('/upload/endpoint')
-    .reply(200)
-    .put('/upload/endpoint')
-    .reply(200);
 
-  const upload = createUploadFixture();
+  const isNumberArraySorted = (a: number[]): boolean => {
+    for (let i = 0; i < a.length - 1; i += 1) {
+      if (a[i] > a[i + 1]) {
+        return false;
+      }
+    }
+    return true;
+  };
+
+  it('progress event fires the correct upload percentage', (done) => {
+    xhrMock.put(endpoint, { status: 200 });
+    const fileBytes = 1048576;
+    const upload = createUploadFixture(
+      {
+        headers: {
+          // NOTE: Adding this as an arbitrary value to cause xhr-mock to cause progress events to be dispatched.
+          // See: https://www.npmjs.com/package/xhr-mock#upload-progress
+          'Content-Length': '1',
+        },
+      },
+      new File([new ArrayBuffer(fileBytes)], 'test.mp4')
+    );
 
-  const successCallback = jest.fn();
+    upload.on('error', (err) => {
+      done(err);
+    });
 
-  upload.on('chunkSuccess', successCallback);
+    let progressCount = 0;
+    const progressArray: number[] = [];
+    upload.on('progress', (progress) => {
+      progressCount = progressCount + 1;
+      progressArray.push(progress.detail);
+    });
 
-  upload.on('success', () => {
-    expect(successCallback).toBeCalledTimes(2);
-    done();
+    upload.on('success', () => {
+      expect(progressCount).to.equal(4);
+      expect(isNumberArraySorted(progressArray)).to.be.true;
+      done();
+    });
   });
-});
 
-const isNumberArraySorted = (a: number[]): boolean => {
-  for (let i = 0; i < a.length - 1; i += 1) {
-    if (a[i] > a[i + 1]) {
-      return false;
-    }
-  }
-  return true;
-};
-
-test('progress event fires the correct upload percentage', (done) => {
-  const fileBytes = 1048576;
-  const upload = createUploadFixture(
-    {},
-    new File([new ArrayBuffer(fileBytes)], 'test.mp4')
-  );
-
-  const scopes = [
-    nock('https://example.com')
-      .matchHeader('content-range', `bytes 0-${fileBytes / 4 - 1}/${fileBytes}`)
```

Comment:
NOTE: I dropped these assertions/expectations from this particular test, since they weren't descriptive of what was being tested here and are already tested in a separate test specifically for these assertions/expectations (See test described as "a file is uploaded using the correct content-range headers")